src/karma_tracker/api/core.clj

```diff
@@ -15,15 +16,16 @@
       (wrap-cors :access-control-allow-origin [#".*"]
                  :access-control-allow-methods [:get])))
 
-(defn api [execution-fns]
-  (-> (routes
-       (query-routes (:query execution-fns)))
-      wrap-handler))
-
 (defn get-execution-fns [resources]
   {:query (partial query/execute resources)})
 
+(defn api [resources]
+  (-> (routes
+       (health-check-routes resources)
+       (query-routes (-> resources get-execution-fns :query)))
+      wrap-handler))
+
 (defn run-server [resources]
   (println "Api server running on port 8000")
-  (run-jetty (api (get-execution-fns resources))
+  (run-jetty (api resources)
```

Comment:
you should add a new execution function (line 20) so that the `api` function doesn't have to know about the resources. They are disconnected in because the `api` just need to know to what functions give control.
This way is easier to test and we don't have one more connection between different part of the app.
I can change it if you want.